algolia/suggestions/client_config.go

```diff
@@ -12,8 +12,11 @@ type IndexConfiguration struct {
 	IndexName string `json:"indexName"`
 	// List of source indices used to generate a Query Suggestions index.
 	SourceIndices []SourceIndex `json:"sourceIndices"`
-	// De-duplicate singular and plural suggestions.
-	Languages []string `json:"languages,omitempty"`
+	// De-duplicate singular and plural suggestions. Can be either a list languages []string or a boolean.
+	// true value means that all the languages are supported.
+	// false value means that singulars and plurals are not considered the same for matching purposes (foot will not find feet).
+	// []string a list of language ISO codes for which singualr and plural suggestions should be enabled.
+	Languages BoolOrStringArray `json:"languages,omitempty"`
```

Comment:
Not sure `omitempty` will work for us here, now that this is no longer nullable: 

```golang
type Foo struct {
	Languages BoolOrStringArray
}

func main() {
	var f Foo
	b, _ := json.Marshal(f)
	fmt.Println(string(b)) // prints {"Languages":null}
}
```

Reply:
True, it doesn't work here. But the engine accepts `null` value for the `languages` field. So I just removed the `omitempty` as it is useless.

Reply:
`omitempty` is used when encoding to json, the engine may accept `null` but do you want `null` to be returned ?

Reply:
If I get it right, even if one explicitly sets `languages` to `null`, the engine always return a non-null  (bool or string array) value, so there is no need to support `null`.